extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/PathParameterValueGettersGenerator.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.resteasy.reactive.links.deployment;
+
+import java.util.Set;
+import java.util.function.BiFunction;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.IndexView;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+import io.quarkus.deployment.util.AsmUtil;
+import io.quarkus.gizmo.DescriptorUtils;
+import io.quarkus.gizmo.Gizmo;
+import io.quarkus.resteasy.reactive.links.runtime.PathParameterInfo;
+
+public class PathParameterValueGettersGenerator extends ClassVisitor {
+
+    private final IndexView index;
+
+    private final String entityType;
+
+    private final Set<PathParameterInfo> pathParameters;
+
+    public static BiFunction<String, ClassVisitor, ClassVisitor> getVisitorFunction(IndexView index, String entityType,
+            Set<PathParameterInfo> pathParameters) {
+        return (className, classVisitor) -> new PathParameterValueGettersGenerator(classVisitor, index, entityType,
+                pathParameters);
+    }
+
+    public PathParameterValueGettersGenerator(ClassVisitor outputClassVisitor, IndexView index, String entityType,
+            Set<PathParameterInfo> pathParameters) {
+        super(Gizmo.ASM_API_VERSION, outputClassVisitor);
+        this.index = index;
+        this.entityType = entityType;
+        this.pathParameters = pathParameters;
+    }
+
+    @Override
+    public void visitEnd() {
```

Comment:
It did initially.
I wasn't sure how to check if a method exists from a visitor and I couldn't use an index, because it doesn't get updated. So I've added an extra mapping here https://github.com/quarkusio/quarkus/pull/14249/files/a6de4a1aa44bc8a1102de72835ba98a1355d5c8d#diff-938ce5408bf74bba0f73f4922b625306f3a9deced50ca5afcde36c03d431f3abR49 to make sure that only one getter is generated.

Reply:
I usually collect signatures in `visitMethod`